index.js

```diff
@@ -39,20 +39,62 @@ class ServerlessCustomDomain {
 
   initializeVariables() {
     if (!this.initialized) {
-      // Sets the credentials for AWS resources.
-      const awsCreds = this.serverless.providers.aws.getCredentials();
-      AWS.config.update(awsCreds);
-      this.apigateway = new AWS.APIGateway();
-      this.route53 = new AWS.Route53();
-      this.setGivenDomainName(this.serverless.service.custom.customDomain.domainName);
-
+      this.enabled = this.evaluateEnabled();
+      if (this.isEnabled) {
+        // Sets the credentials for AWS resources.
+        const awsCreds = this.serverless.providers.aws.getCredentials();
+        AWS.config.update(awsCreds);
+        this.apigateway = new AWS.APIGateway();
+        this.route53 = new AWS.Route53();
+        this.setGivenDomainName(this.serverless.service.custom.customDomain.domainName);
```

Comment:
The above lines are not changed, just indented more, because of the `if` block that now contains them.